### PR TITLE
chore: consolidating spinner color and refactoring institutional qr code modal

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**
Breaking out parts of https://github.com/MetaMask/metamask-extension/pull/24971 for easier review. This PR replaces deprecated colors in favor of the most updated design system colors. I also noticed that this particular component could benefit from some layout and overflow updates. 

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25011?quickstart=1)

## **Related issues**

Part of: https://github.com/MetaMask/metamask-extension/issues/24964

## **Manual testing steps**

1. Go to the storybook build of this page
2. Search for `Spinner` and `QRCodeModal`
3. Check colors have been updated properly in light and dark mode

### **Before**

https://github.com/MetaMask/metamask-extension/assets/8112138/d745a119-f42d-429e-9200-d69b5a1776d1

### **After**

https://github.com/MetaMask/metamask-extension/assets/8112138/282e4460-798a-4841-bad5-242373f7f89b

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/institutional/institutional-components.scss [hunk 1]
```diff
@@ -6,3 +6,4 @@
 **/
 @import 'jwt-dropdown/jwt-dropdown';
 @import 'note-to-trader/index';
+@import 'qr-code-modal/qr-code-modal';
```
Comment (georgewrmarshall) [inline ui/components/institutional/institutional-components.scss:9]: importing stylesheet to main index styles file

File: ui/components/institutional/qr-code-modal/qr-code-modal.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.institutional-qr-code-modal {
+  &__qr-code {
+    background-color: var(--qr-code-white-background);
```
Comment (georgewrmarshall) [inline ui/components/institutional/qr-code-modal/qr-code-modal.scss:3]: Moving inline styles to custom classname

File: ui/components/institutional/qr-code-modal/qr-code-modal.stories.tsx [hunk 3]
```diff
@@ -19,7 +19,7 @@ const testData = {
 const store = mockStore(testData);
 
 const meta: Meta<typeof QRCodeModal> = {
-  title: 'Components/QRCodeModal',
+  title: 'Components/Institutional/QRCodeModal',
```
Comment (georgewrmarshall) [inline ui/components/institutional/qr-code-modal/qr-code-modal.stories.tsx:22]: Fixing storybook title

### Before

Screenshot 2024-06-03 at 1 54 11 PM


### After

Screenshot 2024-06-03 at 1 43 26 PM

File: ui/components/institutional/qr-code-modal/qr-code-modal.tsx [hunk 4]
```diff
@@ -198,63 +203,61 @@ const QRCodeModal: React.FC<QRCodeModalProps> = ({
...
-        {qrCodeValue ? (
-          <Box
-            data-testid="qr-code-visible"
-            style={{
-              padding: 20,
-              backgroundColor: 'var(--qr-code-white-background)',
-              display: 'flex',
-              alignItems: 'center',
-              flexDirection: 'column',
-            }}
```
Comment (georgewrmarshall) [inline ui/components/institutional/qr-code-modal/qr-code-modal.tsx:217]: Removing inlines styles in favor of style utility props and custom classname

File: ui/components/institutional/qr-code-modal/qr-code-modal.tsx [hunk 5]
```diff
@@ -198,63 +203,61 @@ const QRCodeModal: React.FC<QRCodeModalProps> = ({
...
+
+        <ModalBody>
```
Comment (georgewrmarshall) [inline ui/components/institutional/qr-code-modal/qr-code-modal.tsx:213]: ModalBody handles overflow using a scroll container

File: ui/components/institutional/qr-code-modal/qr-code-modal.tsx [hunk 6]
```diff
@@ -198,63 +203,61 @@ const QRCodeModal: React.FC<QRCodeModalProps> = ({
...
+              display={Display.Flex}
+              alignItems={AlignItems.center}
+              flexDirection={FlexDirection.Column}
+              borderColor={BorderColor.borderMuted}
+              borderWidth={1}
+            >
+              <QRCode value={qrCodeValue} size={270} />
+            </Box>
+          ) : (
+            <Spinner />
```
Comment (georgewrmarshall) [inline ui/components/institutional/qr-code-modal/qr-code-modal.tsx:236]: We can remove the color as it now defaults to the correct icon-muted color

File: ui/components/ui/spinner/spinner.component.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-const Spinner = ({ className = '', color = 'var(--color-text-default)' }) => {
+const Spinner = ({ className = '', color = 'var(--color-icon-muted)' }) => {
   return (
```
Comment (georgewrmarshall) [inline ui/components/ui/spinner/spinner.component.js:5]: ### Before


https://github.com/MetaMask/metamask-extension/assets/8112138/e6f07090-c2c9-4a98-929e-5c7879d840ee

### After


https://github.com/MetaMask/metamask-extension/assets/8112138/e3babf7f-3287-49a5-a01a-f6e12db6ccb5

File: ui/components/ui/spinner/spinner.stories.js [hunk 8]
```diff
@@ -13,7 +13,7 @@ export default {
     },
   },
   args: {
-    color: 'var(--color-warning-default)',
+    color: 'var(--color-icon-muted)',
```
Comment (georgewrmarshall) [inline ui/components/ui/spinner/spinner.stories.js:16]: Same as above

**Review by brianacnguyen — DISMISSED** [review]
approved on ds side

**Comment by albertolive** [comment]
Hi @georgewrmarshall,

The refactoring looks great! I really appreciate your work on this.

I have one small request: Could you please reduce the size of the loader? It currently appears quite large, and adjusting it to a smaller size would be perfect.

**Review by albertolive — DISMISSED** [review]

File: ui/components/institutional/qr-code-modal/qr-code-modal.scss [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+  }
+
+  &__spinner {
+    width: 100px;
+  }
```
Comment (georgewrmarshall) [inline ui/components/institutional/qr-code-modal/qr-code-modal.scss:8]: Reducing the size of the spinner

**Comment by georgewrmarshall** [comment]
> Hi @georgewrmarshall,
> 
> The refactoring looks great! I really appreciate your work on this.
> 
> I have one small request: Could you please reduce the size of the loader? It currently appears quite large, and adjusting it to a smaller size would be perfect.

Not a problem! Updated and updated screencasts in description @albertolive

**Review by brianacnguyen — APPROVED** [review]

**Review by albertolive — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**